When using -U, flush the dump file after opening it to force the headers
out immediately.

Suggested by Ferenc Wagner <wferi@niif.hu> in bug #533625.
---
 tcpdump.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/tcpdump.c
+++ b/tcpdump.c
@@ -1205,6 +1205,10 @@
 			callback = dump_packet;
 			pcap_userdata = (u_char *)p;
 		}
+#ifdef HAVE_PCAP_DUMP_FLUSH
+		if (Uflag)
+			pcap_dump_flush(p);
+#endif
 	} else {
 		type = pcap_datalink(pd);
                 printinfo.ndo_type = 1;
